src/DataValues/BoundedQuantityValue.php

```diff
@@ -0,0 +1,381 @@
+		parent::__construct( $amount, $unit );
+
+		if ( $lowerBound->compare( $amount ) > 0 ) {
+			throw new IllegalValueException( '$lowerBound ' . $lowerBound->getValue() . ' must be <= $amount ' . $amount->getValue() );
+		}
+
+		if ( $upperBound->compare( $amount ) < 0 ) {
+			throw new IllegalValueException( '$upperBound ' . $upperBound->getValue() . ' must be >= $amount ' . $amount->getValue() );
+		}
+
+		$this->upperBound = $upperBound;
+		$this->lowerBound = $lowerBound;
+	}
+
+	/**
+	 * Returns a QuantityValue representing the given amount.
+	 * If no upper or lower bound is given, the amount is assumed to be absolutely exact,
+	 * that is, the amount itself will be used as the upper and lower bound.
+	 *
+	 * This is a convenience wrapper around the constructor that accepts native values
+	 * instead of DecimalValue objects.
+	 *
+	 * @note: if the amount or a bound is given as a string, the string must conform
+	 * to the rules defined by @see DecimalValue.
+	 *
+	 * @since 0.1
+	 *
+	 * @param string|int|float|DecimalValue|QuantityValue $number
+	 * @param string $unit A unit identifier. Must not be empty, use "1" for unit-less quantities.
+	 * @param string|int|float|DecimalValue|null $upperBound
+	 * @param string|int|float|DecimalValue|null $lowerBound
+	 *
+	 * @return self
+	 * @throws IllegalValueException
+	 */
+	public static function newFromNumber( $number, $unit = '1', $upperBound = null, $lowerBound = null ) {
+		if ( $number instanceof QuantityValue ) {
+			$number = $number->getAmount();
+		}
+
+		$number = self::asDecimalValue( 'amount', $number );
+		$upperBound = self::asDecimalValue( 'upperBound', $upperBound, $number );
+		$lowerBound = self::asDecimalValue( 'lowerBound', $lowerBound, $number );
+
+		return new self( $number, $unit, $upperBound, $lowerBound );
+	}
+
+	/**
+	 * @see newFromNumber
+	 *
+	 * @deprecated since 0.1, use newFromNumber instead
+	 *
+	 * @param string|int|float|DecimalValue|QuantityValue $number
+	 * @param string $unit
+	 * @param string|int|float|DecimalValue|null $upperBound
+	 * @param string|int|float|DecimalValue|null $lowerBound
+	 *
+	 * @return self
+	 */
+	public static function newFromDecimal( $number, $unit = '1', $upperBound = null, $lowerBound = null ) {
+		if ( $number instanceof QuantityValue ) {
+			$number = $number->getAmount();
+		}
+
+		return self::newFromNumber( $number, $unit, $upperBound, $lowerBound );
+	}
+
+	/**
+	 * @see Serializable::unserialize
+	 *
+	 * @since 0.1
+	 *
+	 * @param string $data
+	 */
+	public function unserialize( $data ) {
+		list( $amount, $unit, $upperBound, $lowerBound ) = unserialize( $data );
+		$this->__construct( $amount, $unit, $upperBound, $lowerBound );
+	}
+
+	/**
+	 * @see DataValue::getType
+	 *
+	 * @since 0.1
+	 *
+	 * @return string
+	 */
+	public static function getType() {
+		return 'quantity'; // FIXME: should this have a different type id? wehat will that break?
```

Comment:
I think: both is the same, the fact that we use subclasses has nothing to do with the data model. It's to make the code nicer.